### PR TITLE
Mizar Pod Networking Design doc for Scaleout Arktos architecture

### PR DESCRIPTION
**Pull request by vinaykul**
Design doc for arktos-mizar-scaleout integartion.

### CODE REVIEW

File: docs/design/mizar_arktos_scaleout.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+
+## Design Options for Arktos Scaleout Architecture
+
+1. Deploy Mizar in each tenant partition as it is deployed in a scaleup cluster.
+  - Pros:
+    - In theory, each tenant would have networking for its own Pods and this is natively isolated from other tenant Pods as long as Geneve VNI is unique.
+    - Pod information available from tenant-local apiserver, therefore little to no changes needed to the Mizar operator.
+  - Cons:
+    - Daemonset support needed in each tenant.
```
Comment (Sindica) [inline docs/design/mizar_arktos_scaleout.md]: Deploy Mizar in each tenant partition is not each tenant, it won't need daemonset support in each tenant.

File: docs/design/mizar_arktos_scaleout.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    - Each Mizar deployment will want to start its own transit daemon. Significant work needed to support multiple transit daemons.
+
+2. Deploy Mizar in each resource partition.
```
Comment (Sindica) [inline docs/design/mizar_arktos_scaleout.md:46]: Current Arktos does not allow Mizar to be deployed in resource partition.

File: docs/design/mizar_arktos_scaleout.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+## Mizar Design Proposal for Scaleout Arktos
+
+This section proposes the following design based on option 3 mentioned above.
+The pros of option 3 outweigh the cons, and it may be implementable without too much development effort.
+
+This design relies on few key changes:
+  - A special Mizar partition that supports daemonset controller to enable Mizar daemon deployment on all nodes.
+  - Mizar operator modified to aggregate nodes from all resource partitions by connecting to all rp apiservers.
+  - A new per-tenant Mizar operator that watches Pods, Services etc created on a particular tenant and creates/deletes endpoints for those objects.
```
Comment (Sindica) [inline docs/design/mizar_arktos_scaleout.md]: new per-tenant-partition operator?

**Review by Sindica — COMMENTED** [review]
Not sure what is the target date of this design. This doc mentioned 2022-01-30 somewhere. I agree that option 3 is the right long term approach but have doubt on the work and targeted release date.

File: docs/design/mizar_arktos_scaleout.md (hunk 1 above)
Comment (vinaykul) [reply docs/design/mizar_arktos_scaleout.md]: fixed in next commit.

File: docs/design/mizar_arktos_scaleout.md (hunk 2 above)
Comment (vinaykul) [reply docs/design/mizar_arktos_scaleout.md:46]: I know. This section only mentions the options considered to inform about the different possible approaches we looked at and their pros and cons.

File: docs/design/mizar_arktos_scaleout.md (hunk 3 above)
Comment (vinaykul) [reply docs/design/mizar_arktos_scaleout.md]: yes. i'll fix it.

**Comment by Sindica** [comment]
Please revise design doc with additional commit. Don't combine historical changes. It is a long document. It is unfriendly for reviewers.

**Comment by vinaykul** [comment]
> Not sure what is the target date of this design. This doc mentioned 2022-01-30 somewhere. I agree that option 3 is the right long term approach but have doubt on the work and targeted release date.

I have updated the doc. This is just the initial design proposal. It is still subject to some more changes as we are doing option 1 at this time. I'll update it as we get new information and our thinking evolves. A living document in that sense..

**Review by phudtran — APPROVED** [review]
LGTM

**Pull request merged**